terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-up.ts

```diff
@@ -206,11 +208,51 @@ async function shouldSkipForRepo(repo: Repo, metrics: Metrics): Promise<boolean>
   return false;
 }
 
-async function allRunnersBusy(
+/**
+ * Returns the maximum number of runners that can be created for the given runner type
+ */
+function getMaximumAllowedScaleUpSize(
+  maxAllowed: number | undefined,
+  provisioned: number,
+  isEphemeral: boolean,
+): number {
+  const NO_LIMIT = Number.MAX_SAFE_INTEGER;
```

Comment:
not a big fan of this indirection. I would rather simply `return Number.MAX_SAFE_INTEGER;`

Reply:
IMO this makes the purpose of the number a bit more understandable